src/librustc_ast_passes/ast_validation.rs

```diff
@@ -561,28 +561,6 @@ impl<'a> AstValidator<'a> {
         }
     }
 
-    /// We currently do not permit const generics in `const fn`,
-    /// as this is tantamount to allowing compile-time dependent typing.
-    ///
-    /// FIXME(const_generics): Is this really true / necessary? Discuss with @varkor.
-    /// At any rate, the restriction feels too syntactic. Consider moving it to e.g. typeck.
```

Comment:
We could certainly move it elsewhere, but there are definitely still concerns with its interactions with `const fn`. At any rate, I don't see that there's any urgent need to combine the two features at present anyway.